src/plugins/intel_npu/tests/functional/behavior/npuw/behavior_tests.cpp

```diff
@@ -53,209 +74,604 @@ TEST_F(BehaviorTestsNPUW, TestInfrastructureIsCorrect) {
+    model = model_generator.get_model_with_one_op();
+
     // Set expectations first:
     {
         InSequence seq;
-        EXPECT_CALL(*npu_plugin, compile_model(A<const std::shared_ptr<const ov::Model>&>(), _)).Times(1);
-        EXPECT_CALL(*cpu_plugin, compile_model(A<const std::shared_ptr<const ov::Model>&>(), _)).Times(1);
+        EXPECT_COMPILE_MODEL(mock_npu, 1);
+        EXPECT_COMPILE_MODEL(mock_cpu, 1);
     }
-    npu_plugin->set_expectations_to_comp_models(0, [](MockCompiledModel& model) {
-        EXPECT_CALL(model, create_sync_infer_request()).Times(1);
-    });
-    npu_plugin->set_expectations_to_infer_reqs(0, [](MockInferRequest& request) {
-        EXPECT_CALL(request, infer())
-            .Times(1)
-            .WillOnce(Throw(std::runtime_error("Infer on MockNPU is failed")));
-    });   
-    cpu_plugin->set_expectations_to_comp_models(0, [](MockCompiledModel& model) {
-        EXPECT_CALL(model, create_sync_infer_request()).Times(1);
-    });
-    cpu_plugin->set_expectations_to_infer_reqs(0, [](MockInferRequest& request) {
-        EXPECT_CALL(request, infer()).Times(1);
-    });
+    EXPECT_CREATE_SYNC_INFER_REQ(mock_npu, MODEL(0), 1);
+    EXPECT_INFER(mock_npu, MODEL(0), 1, THROW("Infer on MockNPU is failed"));
+    EXPECT_CREATE_SYNC_INFER_REQ(mock_cpu, MODEL(0), 1);
+    EXPECT_INFER(mock_cpu, MODEL(0), 1);
 
     // Register mock objects as plugins in OpenVINO:
     register_mock_plugins_in_ov();
 
     // Do the actual test:
-    auto npuw_properties = 
-        ov::AnyMap{ov::intel_npu::use_npuw(true),
-                   ov::intel_npu::npuw::devices("MockNPU,MockCPU")};
-    auto compiled_model = core.compile_model(model, "NPU", npuw_properties);
+    use_npuw_props.emplace(devices("MockNPU,MockCPU"));
+    auto compiled_model = core.compile_model(model, "NPU", use_npuw_props);
     auto infer_request = compiled_model.create_infer_request();
     EXPECT_NO_THROW(infer_request.infer());
 }
 
 TEST_F(BehaviorTestsNPUW, InferIsFailed) {
+    model = model_generator.get_model_with_one_op();
+
+    // Set expectations first:
+    EXPECT_COMPILE_MODEL(mock_npu, 1);
+    EXPECT_COMPILE_MODEL(mock_cpu, 0);
+
+    EXPECT_CREATE_SYNC_INFER_REQ(mock_npu, MODEL(0), 1);
+    EXPECT_INFER(mock_npu, MODEL(0), 1, THROW("Infer on MockNPU is failed"));
+
+    // Register mock objects as plugins in OpenVINO:
+    register_mock_plugins_in_ov();
+
+    // Do the actual test:
+    use_npuw_props.emplace(devices("MockNPU"));
+    auto compiled_model = core.compile_model(model, "NPU", use_npuw_props);
+    auto infer_request = compiled_model.create_infer_request();
+    EXPECT_ANY_THROW(infer_request.infer());
+}
+
+using BehaviorTestsNPUWOnlinePartitioning = BehaviorTestsNPUW;
+TEST_F(BehaviorTestsNPUWOnlinePartitioning, CompilationIsSuccessful) {
+    model = model_generator.get_model_with_repeated_blocks();
+
+    // Set expectations first:
+    EXPECT_COMPILE_MODEL(mock_npu, 12);
+    EXPECT_COMPILE_MODEL(mock_cpu, 0);
```

Comment:
what are these cryptic magic numbers?

Reply:
They are for the "number of times we expect this call"